### PR TITLE
Add .eslintrc

### PR DESCRIPTION
**Pull request by nightwing**
fixes https://github.com/ajaxorg/ace/issues/2746

### CODE REVIEW

**Comment by paladox** [comment]
Why not use jshint instead of eslint since with jshint we could use jscs to make the code look nice.

**Comment by nightwing** [comment]
Because with eslint it was easier to find a config with which Ace code would mostly pass, but maybe i overlooked something in jshint docs.
Does jscs require jshint? It seems to be an independent tool.
Eslint has some code style checking rules as well.

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+lib/ace/mode/javascript/*
+lib/ace/mode/json/*
+lib/ace/mode/lua/*
+lib/ace/mode/php/*
+lib/ace/mode/xml/*
+lib/ace/mode/xquery/*
+lib/ace/mode/xquery.js
+lib/ace/test/asyncjs/
+lib/ace/lib/es5-shim.js
+lib/ace/keyboard/vim*.js
```
Comment (lennartcl) [inline .eslintignore]: Should this also have something like the following?

```
+doc/*
+build/**
+tool/templates/*
```

Running with that I still see 92 errors

**Comment by lennartcl** [comment]
I like it! This will help us maintain code quality.

**Comment by paladox** [comment]
Why not use grunt to run eslint it would be easier. https://github.com/sindresorhus/grunt-eslint

**Comment by nightwing** [comment]
I didn't use grunt/gulp, because didn't want to add several dependencies and a separate file just for calling `eslint "lib/ace/**/*.js"`
But maybe i misunderstood something, @paladox how would it make running eslint easier?

**Comment by paladox** [comment]
Well you would not need to do it through travis ci file you can do it through grunt file that is how it would be made easier.

And I though you doint need grunt/gulp unless that's for recursing into all files and folders.

**Pull request merged**